Casks/font-noto-sans-cjk.rb

```diff
@@ -1,11 +1,16 @@
 cask "font-noto-sans-cjk" do
-  version :latest
-  sha256 :no_check
+  version "2.004"
+  sha256 "a56077736038875f3a5b1e6715b2713a2bd049cc0bf0dfdf645927b070d38bfc"
 
-  url "https://noto-website-2.storage.googleapis.com/pkgs/NotoSansCJK.ttc.zip",
-      verified: "noto-website-2.storage.googleapis.com/"
+  url "https://github.com/notofonts/noto-cjk/releases/download/Sans#{version}/00_NotoSansCJK.ttc.zip"
   name "Noto Sans CJK"
-  homepage "https://www.google.com/get/noto/help/cjk/"
+  desc "Static Super OTC"
+  homepage "https://github.com/notofonts/noto-cjk/tree/main/Sans"
+
+  livecheck do
+    url :url
+    regex(/^Sans(\d+(?:\.\d+)+)$/i)
```

Comment:
```suggestion
    regex(/^Sans[._-]?v?(\d+(?:\.\d+)+)$/i)
    strategy :github_releases
```

The cask `url` is a GitHub release asset, so this should use either `GithubLatest` or `GithubReleases`. Upstream creates alternating Serif and Sans releases and the "latest" release may be either, so we have to use `GithubReleases` here.

Past that, this suggestion adds some optional components to the regex, as there's some variety in their past tag format. Hopefully they will stick with the `Sans2.004` format going forward, though.

Reply:
> Upstream creates alternating Serif and Sans releases and the "latest" release may be either, so we have to use `GithubReleases` here.

Ah thanks! Then I guess all the `font-noto-serif-cjk-*` casks need similar strategy. Do they?



> Past that, this suggestion adds some optional components to the regex, as there's some variety in their past tag format.

You meant tags in other repositories of notofonts organisation? Because it seems for `notofonts/noto-cjk` the variety in tag formats is not to be matched by the updated regex:

```
$ gh release list --repo notofonts/noto-cjk

TITLE                                           TYPE    TAG NAME        PUBLISHED
Noto Serif CJK Version 2.002 (OTF, OTC, Sup...  Latest  Serif2.002      about 3 months ago
Noto Sans CJK Version 2.004 (OTF, OTC, Supe...          Sans2.004       about 1 year ago
Noto Serif CJK Version Serif2.001 (OTF, OTC...          Serif2.001      about 1 year ago
Noto CJK v20201206 as of 2020.12.06                     v20201206-cjk   about 2 years ago
Noto Sans CJK V2.001                                    NotoSansV2.001  about 4 years ago
Match Adobe's 1.004 release that corrects v...          v1.004          about 8 years ago
```

Reply:
> Then I guess all the `font-noto-serif-cjk-*` casks need similar strategy

Since the files in the variant casks are provided as part of the same release as `font-noto-sans-cjk`, we can simply use `cask "font-noto-sans-cjk"` in those `livecheck` blocks to avoid the duplication. [We only use `#cask` when there's a clear primary/secondary relationship between the casks. In this cask `font-noto-sans-cjk` is the primary cask and these are secondary variants, so it makes sense here.]

I'll push a commit to that branch in a moment to update those `livecheck` blocks but we'll need to rebase that PR after this one is merged to use the `livecheck` block from this PR.

> You meant tags in other repositories of notofonts organisation? Because it seems for `notofonts/noto-cjk` the variety in tag formats is not to be matched by the updated regex

I was referring to the tags in the notofonts/noto-cjk repository, as you've listed. `/^Sans[._-]?v?(\d+(?:\.\d+)+)$/i` will match tags like `Sans2.004`, `SansV2.004`, `Sans-v2.004`, etc. I deliberately omitted an optional leading `(?:Noto)?` for now, so it won't match the older `NotoSansV2.001` (which doesn't have release assets). That said, there's no harm in matching the older release (with 2.004 being newer than it). I'll update the suggestion above.

Reply:
> we can simply use `cask "font-noto-sans-cjk"` in those `livecheck` blocks to avoid the duplication.

Thank you again! Now I know something new about `livecheck`.
https://docs.brew.sh/Brew-Livecheck#referenced-formulacask.

> I've pushed a commit to that branch to update those `livecheck` blocks but we'll need to rebase that PR after this one is merged.

I'll open another PR updating `livecheck` for Noto Serif CJK cask and its variants (serif, not sans), which were updated previously in #8720 and #8721.